api.go

```diff
@@ -0,0 +1,265 @@
+package metaphor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type MetaphorClient struct {
+	apiKey      string
+	RequestBody *RequestBody
+}
+
+type RequestBody struct {
+	Query              string   `json:"query,omitempty"`
+	Url                string   `json:"url,omitempty"`
+	NumResults         int      `json:"numResults,omitempty"`
+	IncludeDomains     []string `json:"includeDomains,omitempty"`
+	ExcludeDomains     []string `json:"excludeDomains,omitempty"`
+	StartCrawlDate     string   `json:"startCrawlDate,omitempty"`
+	EndCrawlDate       string   `json:"endCrawlDate,omitempty"`
+	StartPublishedDate string   `json:"startPublishedDate,omitempty"`
+	EndPublishedDate   string   `json:"endPublishedDate,omitempty"`
+	UseAutoprompt      bool     `json:"useAutoprompt,omitempty"`
+	Type               string   `json:"type,omitempty"`
+}
+
+const (
+	// DEFAULT REQUEST VALUES
+
+	// DefaultNumResults is the default number of expected results
+	DefaultNumResults = 10
+	// DefaultAutoprompt if true, your query will be converted to a Metaphor query.
+	// If findLinks ednpoint is used needs to be nil to omit useAutoprompt field from RequestBody
+	DefaultAutoprompt = false
+	// DefaultSearchType
+	DefaultSearchType = "neural"
+
+	// DEFAULT API ENDPOINT URL's
+
+	// DefaultSearchURL is the default search endpoint
+	DefaultSearchURL = "https://api.metaphor.systems/search"
```

Comment:
would be nice to have a base URL here and possible to instantiate the client using it - that way, people can mock more easily